### PR TITLE
return the test to replace transaction back

### PR DESCRIPTION
**Pull request by anastasiyaig**
return the test to replace transaction back

### CODE REVIEW

File: tests/test_block_creation_mvp.py [hunk 1]
```diff
@@ -225,3 +225,36 @@ def test_transaction_wrong_payload(self, transaction_template, data_set):
...
+                                 self.receiver_address,
+                                 self.sender_private_key))
+        # TODO: maybe add a verification that transaction is in pool
+
+        initial_txn_nonce = self.web3.eth.get_transaction(initial_txn_hash)["nonce"]
+
+        raw_txn = {
+            "from": self.sender_address,
+            "gasPrice": self.web3.eth.gas_price,
+            "gas": 6000000,
```
Comment (Serhy) [inline tests/test_block_creation_mvp.py]: Thank you @anastasiyaig 
Good idea of replacing transaction.
However, we missed with param a little. Currently you vary `gas` value. But we need to decrease `gasPrice` value (like it says in the issue #5214 ) in order transaction is hang in tx_pool, and then with subsequent transaction with high enough gasPrice - to replace the transaction.

**Review by Serhy — CHANGES_REQUESTED** [review]

File: tests/test_block_creation_mvp.py (hunk 1 above)
Comment (anastasiyaig) [reply tests/test_block_creation_mvp.py]: okay,  i will fix it

Comment (anastasiyaig) [reply tests/test_block_creation_mvp.py]: @Serhy i have updated the logic of the test , so hopefully now it uses correct parameter (`gasPrice`). I am stuck on debugging why the test is failing on the verification for transaction hash of new transaction (with higher `gasPrice`). The assertion error is raised that this transaction has not been found in transaction pool. Either the `check_transaction_is_in_pool` method is not correct, or this replacing transaction is not sent to transaction pool. Need some help here.

File: tests/test_block_creation_mvp.py [hunk 2]
```diff
@@ -225,3 +225,27 @@ def test_transaction_wrong_payload(self, transaction_template, data_set):
...
+
+        assert low_gas_price_txn_hash in str(self.web3.geth.txpool.content())
+
+        transaction_template_low_gas_price.update(
+            {"gasPrice": int(self.web3.eth.gas_price * 3),
+             "nonce": low_gas_price_txn_nonce})
+
+        replacing_txn_hash = self.send_transaction(transaction_template_low_gas_price)
+
+        assert replacing_txn_hash in str(self.web3.geth.txpool.content())
```
Comment (Serhy) [inline tests/test_block_creation_mvp.py]: We need to check that first transaction (low_gas_price_txn_hash) is not in the pool now (replaced).
And, replacing_txn_hash - should be mined and included into block. So let's check atthe end that replacing_txn_hash is mined and no ((1) low_gas_price_txn_hash AND (2) replacing_txn_hash) in the tx_pool

**Review by Serhy — CHANGES_REQUESTED** [review]

**Comment by Serhy** [comment]
Thank you @anastasiyaig 
It's ready for merge now.

**Review by Serhy — APPROVED** [review]

**Pull request merged**